DB/src/main/java/io/deephaven/db/v2/parquet/ParquetInstructions.java

```diff
@@ -0,0 +1,64 @@
+package io.deephaven.db.v2.parquet;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ParquetInstructions {
+    private Map<String, String> columnNameMapping;
+    private Map<String, String> reverseColumnNameMapping;
+
+    public ParquetInstructions() {
+    }
+
+    public String getColumnNameMapping(final String fromName) {
+        if (columnNameMapping == null) {
+            return null;
+        }
+        return columnNameMapping.get(fromName);
+    }
+
+    public Map<String, String> getColumnNameMappings() {
+        if (columnNameMapping == null) {
+            return Collections.emptyMap();
+        }
+        return columnNameMapping;
+    }
+
+    public Map<String, String> getReverseColumnNameMappings() {
+        if (reverseColumnNameMapping == null) {
+            return Collections.emptyMap();
+        }
+        return reverseColumnNameMapping;
+    }
+
+    public ParquetInstructions addColumnNameMapping(final String fromName, final String toName) {
+        if (columnNameMapping == null) {
+            columnNameMapping = new HashMap<>();
+            reverseColumnNameMapping = new HashMap<>();
+        }
+        final String previousFrom = reverseColumnNameMapping.get(toName);
+        if (previousFrom != null && !previousFrom.equals(fromName)) {
```

Comment:
I see why you wanted to use maps, now. Maybe we only need to support one-way remapping (from parquet to deephaven), and we could pass this class's `getColumnNameMapping` method as the mapper, and have it record reverse mappings for validation as it's used?

Reply:
See comment above (the one where I am quoting Boog).

Reply:
Re-re-done.